dojo/jira_link/helper.py

```diff
@@ -131,34 +131,35 @@ def get_jira_url(obj):
     logger.debug('getting jira url')
 
     # finding + engagement
-    if hasattr(obj, 'has_jira_issue') and obj.has_jira_issue:
-        return get_jira_issue_url(obj)
-    else:
+    issue = get_jira_issue(obj)
+    if issue is not None:
+        return get_jira_issue_url(issue)
+    elif isinstance(obj, Finding):
         # finding must only have url if there is a jira_issue
         # engagement can continue to show url of jiraproject instead of jira issue
-        if isinstance(obj, Finding):
-            return None
+        return None
 
     if isinstance(obj, JIRA_Project):
         return get_jira_project_url(obj)
 
     return get_jira_project_url(get_jira_project(obj))
 
 
-def get_jira_issue_url(obj):
+def get_jira_issue_url(issue):
     logger.debug('getting jira issue url')
-    if obj.has_jira_issue:
-        jira_project = get_jira_project(obj)
-        jira_instance = get_jira_instance(obj)
-        if jira_project and jira_instance:
-            # example http://jira.com/browser/SEC-123
-            return jira_project.jira_instance.url + '/browse/' + obj.jira_issue.jira_key
+    jira_project = issue.jira_project
+    if jira_project is None:
+        return None
 
-    return None
+    jira_instance = jira_project.jira_instance
```

Comment:
please leave the call to `get_jira_project` and `get_jira_instance` in place instead of accessing properties deirectly.